src/nexgen/nxs_utils/Source.py

```diff
@@ -0,0 +1,88 @@
+"""
+Object definition for Source, Beam and Attenuator
+"""
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from dataclasses_json import dataclass_json
+
+from ..utils import Facility
+
+
+class Source:
+    def __init__(
+        self,
+        beamline: str,
+        facility: Facility = Facility(
+            "Diamond Light Source",
+            "DLS",
+            "Synchrotron X-ray Source",
+            None,
+        ),
+        probe: str | None = None,
+    ):
+        self.beamline = beamline
+
+        self.name = facility.name
+        self.short_name = facility.short_name
+        self.facility_type = facility.type
+        self.facility_id = facility.id
+
+        self.probe = probe
+
+    def __repr__(self) -> str:
+        msg = f"Beamline {self.beamline} at {self.name}.\n"
+        if self.probe:
+            msg += "Probe: {}"
+        return msg
+
+    def set_instrument_name(self) -> str:
+        """Set the instrument name from the details saved in source.
+
+        If source type is not defined a priori, the function will assume it is a Synchrotron.
+        If the facility_id is defined inside the source dictionary, that is the value that will be used.
+        Naming tries to follow the recommended convention for NXmx:
+        https://mmcif.wwpdb.org/dictionaries/mmcif_pdbx_v50.dic/Items/_diffrn_source.type.html
+
+        Returns:
+            name (str): The name to write under '/entry/instrument/name'
+        """
+        facility_id = "DIAMOND" if self.facility_id is None else self.facility_id
```

Comment:
Why not just set the default id to `DIAMOND` above?

Reply:
Both this and the previous one are setting up for a couple of requests for Electron Diffraction metadata... refactoring that is still a wip.